client/gui/CGuiHandler.cpp

```diff
@@ -338,22 +356,53 @@ void CGuiHandler::handleCurrentEvent()
 			it->textEdited(current->edit);
 		}
 	}
-	//todo: muiltitouch
 	else if(current->type == SDL_MOUSEBUTTONUP)
 	{
-		switch(current->button.button)
+		if(multifinger && lastFingerCount == 1)
 		{
-		case SDL_BUTTON_LEFT:
-			handleMouseButtonClick(lclickable, EIntObjMouseBtnType::LEFT, false);
-			break;
-		case SDL_BUTTON_RIGHT:
+			multifinger = false;
+		}
+		else
+		{
+			switch (current->button.button)
+			{
+			case SDL_BUTTON_LEFT:
+				handleMouseButtonClick(lclickable, EIntObjMouseBtnType::LEFT, false);
+				break;
+			case SDL_BUTTON_RIGHT:
+				handleMouseButtonClick(rclickable, EIntObjMouseBtnType::RIGHT, false);
+				break;
+			case SDL_BUTTON_MIDDLE:
+				handleMouseButtonClick(mclickable, EIntObjMouseBtnType::MIDDLE, false);
+				break;
+			}
+		}
+	}
+	else if(current->type == SDL_FINGERDOWN)

```

Comment:
Should we allow this section only for ANDROID or is it ok to live it for all devices?

Reply:
I do not see any harm if iOS will also support 2-finger rclick

Reply:
I can test on iOS

Reply:
on iOS doesn't work properly - impossible to close pop-up windows.
Long touch works well on iOS, so I'd recommend to disable double touch as right click for iOS, but maybe reuse it as "space" behavior

Reply:
Maybe it is similar to windows simulator issue. Do you remove both fingers at once?

Reply:
Hope, I managed to fix it. Can you try one more time?

Reply:
It was even reproducible on Android. You just need to remove fingers quick enough.